src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java

```diff
@@ -78,4 +104,42 @@ public void doPostCleanup(BibEntry entry) {
     public Parser getParser() {
         return new BibtexParser(importFormatPreferences, new DummyFileUpdateMonitor());
     }
+
+
+    public List<BibEntry> performSearch(BibEntry entry) {
+        List<BibEntry> results = new ArrayList<>();
+
+        String doi = entry.getField(StandardField.DOI).orElse(null);
+        if (doi == null) {
+            return results;
+        }
+
+        String url = "https://inspirehep.net/api/doi" + doi;
+
+        try {
+            URL obj = new URL(url);
```

Comment:
You can  use our UrlDownload class for this
  
   URLDownload download = getUrlDownload(doiURL);
                download.addHeader("Accept", MediaTypes.APPLICATION_BIBTEX);
                
     This also handles errors

Reply:
@Siedlerchr I have fixed some logical bugs and corrected code style.

Now InspireHEP is working as desired as one of the resources to fetch bibliographic data from when given a valid doi.

![Screenshot_20230214_235455](https://user-images.githubusercontent.com/114904959/218826620-0d46ffec-c23b-4cb3-b2d8-43fa0695b0e4.png)